plugins/scale/scale-title-overlay.cpp

```diff
@@ -0,0 +1,326 @@
+        overflow = res.width > overlay.tex.width;
+        view->damage();
+    }
+
+  public:
+    enum class position
+    {
+        TOP,
+        CENTER,
+        BOTTOM,
+    };
+
+    position pos = position::CENTER;
+
+    GLuint get_texture(float output_scale, wf::geometry_t& geometry)
+    {
+        wlr_box box = get_transformed_bounding_box();
+        int w = overlay.tex.width;
+        int h = overlay.tex.height;
+
+        /**
+         * regenerate the overlay texture in the following cases:
+         * 1. Output's scale changed
+         * 2. The overlay does not fit anymore
+         * 3. The overlay previously did not fit, but there is more space now
+         * TODO: check if this wastes too high CPU power when views are being
+         * animated and maybe redraw less frequently or in a way that will not
+         * hold up the rendering loop!
+         */
+        if ((output_scale != par.output_scale) || (w > box.width * output_scale) ||
+            (overflow && (w < std::floor(box.width * output_scale))))
+        {
+            par.output_scale = output_scale;
+            par.max_size     = {box.width, box.height};
+            auto res = overlay.render_text(view->get_title().c_str(), par);
+            view->damage();
+            w = overlay.tex.width;
+            h = overlay.tex.height;
+            overflow = res.width > w;
+        }
+
+        int y = 0;
+        switch (pos)
+        {
+          /* TODO: Add some padding! */
+          case position::TOP:
+            y = box.y - (int)(h / output_scale);
+            break;
+
+          case position::CENTER:
+            y = box.y + box.height / 2 - (int)(h / output_scale / 2);
+            break;
+
+          case position::BOTTOM:
+            y = box.y + box.height;
+            break;
+        }
+
+        geometry = {box.x + box.width / 2 - (int)(w / output_scale / 2),
+            y, (int)(w / output_scale), (int)(h / output_scale)};
+        return overlay.tex.tex;
+    }
+
+    explicit view_title_overlay(wayfire_view v, wf::scale_transformer& tr_,
+        int font_size,
+        const wf::color_t& bg_color, const wf::color_t& text_color) : view(v),
+        tr(tr_)
+    {
+        par.font_size  = font_size;
+        par.bg_color   = bg_color;
+        par.text_color = text_color;
+        par.exact_size = true;
+        render_overlay();
+        view->connect_signal("title-changed", &view_changed);
+    }
+
+    wf::signal_connection_t view_changed{[this] (auto)
+        {
+            render_overlay();
+        }
+    };
+
+    wf::dimensions_t get_size() const
+    {
+        return {overlay.tex.width, overlay.tex.height};
+    }
+};
+
+
+scale_show_title::scale_show_title() :
+    view_filter{[this] (auto)
+    {
+        update_title_overlay_opt();
+    }},
+
+    scale_end{[this] (wf::signal_data_t*)
+    {
+        show_view_title_overlay = title_overlay_t::NEVER;
+        last_title_overlay = nullptr;
+        if (mouse_update_connected)
```

Comment:
you can just disconnect, it is a no-op if the signal is not connected (and in this case, no need for `mouse_update_connected`)